programs/perpetuals/src/state/custody.rs

```diff
@@ -214,6 +214,24 @@ impl Custody {
             && self.borrow_rate.validate()
     }
 
+    pub fn get_cumulative_interest(&self, curtime: i64) -> Result<u128> {
```

Comment:
have you thought about how this might affect the slope and the base rate? It makes sense to do it as the loan is perpetual.

Reply:
the purpose of this function is to get interest accumulated since the last borrow rate change, i.e. it doesn't alter the slope nor affected by it

Reply:
No I get that. What I was getting at, and is probably not straight forward to hypothesis just now, is that as you move to compounded borrow rates the parameters of the borrowing rate stepwise function should probably be different than the ones used by GMX.

Reply:
I actually use simple interest rate not compounded, as is in A = P(1+rt), not A = P(1+r)^t. Not sure what GMX uses. Compounded grows exponentially discouraging traders from holding positions for too long. But maybe thats ok. I was thinking to add a configurable parameter to switch between simple/compounded. The only problem with the latter is calculations can overflow quickly, I'll need to think about the math. One solution is to use floats, but it is slow. It might look something like this:
```
	pub fn get_cumulative_interest(&self, curtime: i64) -> Result<u128> {
		if curtime > self.borrow_rate_state.last_update {
			let elapsed_time_sec =
				math::checked_sub(curtime, self.borrow_rate_state.last_update)? as u128;

			match self.borrow_rate.interest_type {
				InterestType::Simple => {
					let cumulative_interest = math::checked_ceil_div(
						math::checked_mul(
							elapsed_time_sec,
							self.borrow_rate_state.current_rate as u128,
						)?,
						3600,
					)?;
					math::checked_add(
						self.borrow_rate_state.cumulative_interest,
						cumulative_interest,
					)
				}
				InterestType::Compound => {
					let cumulative_interest_float = math::checked_powi(
						math::checked_add(
							1,
							math::checked_float_div(
								self.borrow_rate_state.current_rate as f64,
								math::checked_float_mul(3600, Perpetuals::RATE_POWER)?,
							)?,
						)?,
						elapsed_time_sec,
					)?;
					let cumulative_interest =
						math::checked_mul(cumulative_interest_float, Perpetuals::RATE_POWER)?;

					(self.borrow_rate_state.cumulative_interest + 1) * cumulative_interest - 1
				}
			}
		} else {
			Ok(self.borrow_rate_state.cumulative_interest)
		}
```
Feel free to contribute if you have some ideas.